### PR TITLE
fix Issue 15782 - Alias no longer strips qualifiers from user defined types

### PR DESCRIPTION
**Pull request by MartinNowak**

### CODE REVIEW

**Comment by JakobOvrum** [comment]
LGTM. Ping @John-Colvin

**Comment by MartinNowak** [comment]
We still need to fix https://github.com/D-Programming-Language/dmd/pull/5302#issuecomment-198815649 in order for the message to show.
People might trip over this when using sth. like [Replace](https://dlang.org/phobos/std_meta.html#.Replace), I hope the message is clear enough.

**Comment by 9rnsr** [comment]
I think the new publice `Alias` and old internal `Alias` uses should be separated, and deprecation messages need to be shown only for the latter.

File: std/meta.d [hunk 1]
```diff
@@ -60,6 +60,9 @@ unittest
     static assert(is(Types == AliasSeq!(int, double, char)));
 }
 
+private enum isUDT(T) = is(T == struct) || is(T == class) || is(T == interface);
```
Comment (rcorre) [inline std/meta.d]: does this need to handle `union` as well? `isUDT` look like `isAggregateType` minus `(T == union)`.

**Comment by MartinNowak** [comment]
Updated

**Comment by 9rnsr** [comment]
LGTM.

**Comment by 9rnsr** [comment]
Auto-merge toggled on

**Comment by MartinNowak** [comment]
> Auto-merge toggled on

Unfortunately the deprecation messages don't work, e.g. when using `staticIndexOf`, but we can figure that out for the next release.

**Pull request merged**